### ISSUE TITLE
GTIN not decoded correctly from EPC

### ISSUE THREAD

**Issue by legomatrix** [fulfilio/pyepc#4]
* Python version: Python 3.8.10
* Operating System: Ubuntu

### Description

GS1 site decodes this GTIN 08719139005740 from the following EPC
3036142C8C008F8000053244
Via pyepc we get 0871913900574**1**0 as a GTIN

Can someone explain?
Have no idea where this extra **1** is coming from.

### What I Did
`epcs = [x for x in df[1]]

for x in epcs:
  epc = decode(x)`

### Resources
https://www.gs1.org/services/epc-encoderdecoder

**Comment by NiharZanwar** [comment]
according to me the bug is in this function https://github.com/fulfilio/pyepc/blob/e72091736f45afe0c09b88233a7694751734dd3e/pyepc/utils.py#L136 


```
def calculate_check_digit(number):
    """
    Given a number without the check-digit, calculate
    the check digit and return it.

    See: https://www.gs1.org/services/how-calculate-check-digit-manually
    """
    # Step 2: Multiply alternate position by 3 and 1
    # and get the sum
    step_2 = sum(
        [int(n) * 3 for n in number[::-1][::2]]
    ) + sum(
        [int(n) for n in number[::-1][1::2]]
    )

    # Subtract the sum from nearest equal or higher multiple of ten
    return str(10 - (step_2 % 10))
```
`# Subtract the sum from nearest equal or higher multiple of ten` this step is not followed if `step_2 % 10 == 0`

suggested fix 
```
def calculate_check_digit(number):
    """
    Given a number without the check-digit, calculate
    the check digit and return it.

    See: https://www.gs1.org/services/how-calculate-check-digit-manually
    """
    # Step 2: Multiply alternate position by 3 and 1
    # and get the sum
    step_2 = sum(
        [int(n) * 3 for n in number[::-1][::2]]
    ) + sum(
        [int(n) for n in number[::-1][1::2]]
    )
    if step_2 % 10 == 0:
        return "0"
    # Subtract the sum from nearest equal or higher multiple of ten
    return str(10 - (step_2 % 10))
```

**Comment by legomatrix** [comment]
Had some issues upgrading via pip to the new code but the issue is indeed fixed now. 
Perfect, thanks @NiharZanwar  and @sharoonthomas.  🥳 

3036142C8C013CC0002B1B71 is another test case if you want. it should decode to 08719139012670 and **not** 0871913901267**1**0

**Comment by NiharZanwar** [comment]
Funny story - I remember a year ago I had spent 8 hrs trying to debug the issue in my code ( A little background here - I am CTO at TagID Solutions India and we are developing RFID based inventory management solutions for the apparel retail ). I sat there debugging this issue in a warehouse where our RFID tunnel was not working. I thought the issue was with my code somewhere but then fortunately I found the bug. :partying_face:  :champagne: .
